price-feeder/config/config.go

```diff
@@ -216,9 +218,17 @@ func ParseConfig(configPath string) (Config, error) {
 		}
 	}
 
+	// Use currencyProviderTracker to check available exchanges for each currency pair
+	// and determine minimum amount of providers required for each pair
+	currencyProviderTracker, err := NewCurrencyProviderTracker(ctx, logger, cfg.CurrencyPairs...)
+	if err != nil {
+		return cfg, fmt.Errorf("failed to start currency provider tracker: %w", err)
+	}
+
 	for base, providers := range pairs {
-		if _, ok := pairs[base][provider.ProviderMock]; !ok && len(providers) < 3 {
-			return cfg, fmt.Errorf("must have at least three providers for %s", base)
+		minProviders := currencyProviderTracker.getCurrencyProviderMin(base)
```

Comment:
Instead of having to call functions on currencyProviderTracker here you could return the entire `map[string]int` initially from the currencyProviderTracker. It would turn currencyProviderTracker into more of a service architecture where you pass in currencyPairs and get back a map of the min providers for each one.

This check could probably be moved outside of ParseConfig to reduce how intertwined the responsibilities are.

Reply:
I've removed the getter methods now and the check into another method.